scripts/partition/txt.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Command-line tool to visualize StreamingDataset sample space partitioning."""
+
+import math
+from argparse import ArgumentParser, Namespace
+
+from streaming.base.partitioning import get_partitions
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('-n', '--dataset_size', type=int, default=678)
+    args.add_argument('-b', '--device_batch_size', type=int, default=7)
+    args.add_argument('-o', '--offset_in_epoch', type=int, default=0)
+    args.add_argument('-c', '--canonical_nodes', type=int, default=6)
+    args.add_argument('-p', '--physical_nodes', type=int, default=3)
+    args.add_argument('-d', '--node_devices', type=int, default=4)
+    args.add_argument('-w', '--device_workers', type=int, default=5)
+    return args.parse_args()
+
+
+def main(args: Namespace) -> None:
+    """Generate and display a partitioning given command-line arguments.
+
+    Args:
+        args (Namespace): Command-line arguments.
+    """
+    ids = get_partitions(args.dataset_size, args.canonical_nodes, args.physical_nodes,
```

Comment:
probably not critical to have nice error messages for everything since this is a debugging script mostly, but when i set `--offset_in_epoch 100` and all other default args, it crashed because the ids can't be reshaped
```
  File "/Users/danielking/miniconda3/envs/composer/lib/python3.10/site-packages/streaming/base/partitioning.py", line 68, in get_partitions
    x = x.reshape(-1, num_physical_nodes)
ValueError: cannot reshape array of size 578 into shape (3)
```

Not sure if there are other incompatible argument combinations hiding.

Reply:
Nice find. That should be the only one.